File: 15.js
function countCollisions(values, size) {
    const table = new Array(size).fill(null);
    let collisions = 0;

    for (let v of values) {
        let idx = v % size;
        while (table[idx] !== null) {
            collisions++;
            idx = (idx + 1) % size;
        }
        table[idx] = v;
    }
    return collisions;
}

console.log(countCollisions([12, 22, 32, 42], 10));

File: 15.py
def count_collisions(values, size):
    table = [None] * size
    collisions = 0

    for v in values:
        idx = v % size
        while table[idx] is not None:
            collisions += 1
            idx = (idx + 1) % size
        table[idx] = v

    return collisions

print(count_collisions([12, 22, 32, 42], 10))
